api/mysagw/identity/filters.py

```diff
@@ -244,15 +244,23 @@ def filter_queryset(self, request, queryset, view):
                 filters.append(condition)
 
         needed = [set(queryset.filter(f).values_list("pk", flat=True)) for f in filters]
-        needed_pks = reduce(lambda a, b: a.intersection(b), needed)
```

Comment:
Hmm.. What was the reason why we didn't do this on DB level?

I feel like (without knowing the history) that we could do this with something like this:

```python
if filters:
    queryset = queryset.filter(reduce(operator.and_, filters))
```

Reply:
Tbh, I don't really remember. But we had crazy performance problems with the first implementation and it took a long pairing session with you, to make it performant.